src/components/nodes/LabelUpdaterNode.tsx

```diff
@@ -0,0 +1,110 @@
+import { CloseIcon } from "@chakra-ui/icons";
+import { Box, Input } from "@chakra-ui/react";
+import { FormEvent, useEffect, useRef, useState } from "react";
+import { Handle, Position, useReactFlow } from "reactflow";
+import { Row } from "../../utils/chakra";
+import { modifyFluxNodeLabel, modifyFluxNode } from "../../utils/fluxNode";
+import { FluxNodeData } from "../../utils/types";
+
+export function LabelUpdaterNode({
+  id,
+  data,
+  isConnectable,
+}: {
+  id: string;
+  data: FluxNodeData;
+  isConnectable: boolean;
+}) {
+  const { setNodes } = useReactFlow();
+
+  const [renameLabel, setRenameLabel] = useState(data.label);
+  const renameBlockRef = useRef<HTMLInputElement>(null);
+
+  const cancel = () => {
+    setNodes((nodes) =>
+      modifyFluxNode(nodes, {
+        id,
+      })
+    );
+  };
+
+  const renameNode = () => {
+    setNodes((nodes) =>
+      modifyFluxNodeLabel(nodes, {
+        id,
+        label: renameLabel,
+      })
+    );
+  };
+
+  const submit = (e: FormEvent) => {
+    e.preventDefault();
+    renameNode();
+  };
+
+  useEffect(() => {
+    function handleClick(event: any) {
+      if (renameBlockRef.current && !renameBlockRef.current.contains(event.target)) {
+        cancel();
+      }
+    }
+    function handleClose(event: any) {
+      if (event.key === "Escape") cancel();
+    }
+
+    // Bind the event listeners
+    document.addEventListener("mousedown", handleClick);
+    document.addEventListener("keydown", handleClose);
+    return () => {
+      // Unbind the event listeners on clean up
+      document.removeEventListener("mousedown", handleClick);
+      document.removeEventListener("keydown", handleClose);
+    };
+  }, [renameBlockRef]);
+
+  useEffect(() => {
+    // select the input element on mount
+    const input = renameBlockRef.current?.querySelector("input");
+    if (input) {
+      input.select();
+    }
```

Comment:
let's just give the input an id and select it that way, tryna get rid of refs here hehe

Reply:
wellll now that we're gonna leave the ref for the clicking logic maybe we leave this? i know im just microptimizing here but i cant resist, do we know if using ref.current.queryselector would be faster than querying id from window?